src/cr/cube/matrix/cubemeasure.py

```diff
@@ -197,10 +197,19 @@ def factory(cls, cube, dimensions, slice_idx):
             raise ValueError(
                 "cube-result does not contain cube-valid-overlaps measure"
             )  # pragma: no cover
-        return _CatXMrOverlaps(
-            dimensions,
-            cube.overlaps[cls._slice_idx_expr(cube, slice_idx)],
-            cube.valid_overlaps[cls._slice_idx_expr(cube, slice_idx)],
+        dimension_types = tuple(d.dimension_type for d in dimensions)
+        return (
+            _MrXMrOverlaps(
+                dimensions,
+                cube.overlaps[cls._slice_idx_expr(cube, slice_idx)],
+                cube.valid_overlaps[cls._slice_idx_expr(cube, slice_idx)],
+            )
+            if dimension_types == (DT.MR, DT.MR)
+            else _CatXMrOverlaps(
+                dimensions,
+                cube.overlaps[cls._slice_idx_expr(cube, slice_idx)],
+                cube.valid_overlaps[cls._slice_idx_expr(cube, slice_idx)],
+            )
```

Comment:
I would use a variable like `slice_idx_expr = cls._slice_idx_expr(cube, slice_idx)`, and use it in the 4 slicing operations at line 204,205,210,211